HerPublicWebsite.BusinessLogic/QuestionnaireUpdater.cs

```diff
@@ -126,16 +130,30 @@ public async Task<Questionnaire> GenerateReferralAsync(Questionnaire questionnai
             );
         }
 
+        try
+        {
         var perReferralReport = new PerReferralReport(referralRequest);
         await dataAccessProvider.PersistPerReferralReportAsync(perReferralReport);
+        }
+        catch (Exception e)
+        {
+            logger.LogError("Couldn't generate per referral report: {}", e.Message);
+        }
 
         return questionnaire;
     }
 
     public async Task<Questionnaire> GenerateAnonymisedReportAsync(Questionnaire questionnaire)
+    {
+        try
     {
         var anonymisedReport = new AnonymisedReport(questionnaire);
         await dataAccessProvider.PersistAnonymisedReportAsync(anonymisedReport);
+        }
+        catch (Exception e)
+        {
+            logger.LogError("Couldn't generate anonymised report: {}", e.Message);
+        }

```

Comment:
Not sure if it's just github, but indenting looks odd here

Reply:
Yep something went wrong there, that's fixed now